devito/ir/support/properties.py

```diff
@@ -0,0 +1,73 @@
+from devito.tools import Tag
+
+
+class Property(Tag):
+
+    _KNOWN = []
+
+    def __init__(self, name, val=None):
+        super(Property, self).__init__(name, val)
+        Property._KNOWN.append(self)
+
+
+SEQUENTIAL = Property('sequential')
+"""A sequential Dimension."""
+
+PARALLEL = Property('parallel')
+"""A fully parallel Dimension."""
+
+PARALLEL_IF_ATOMIC = Property('parallel_if_atomic')
+"""A parallel Dimension with local reductions."""
+
+COLLAPSED = lambda i: Property('collapsed', i)
+"""Collapsing Dimensions."""
+
+VECTORIZED = Property('vector-dim')
+"""A SIMD-vectorized Dimension."""
+
+TILABLE = Property('tilable')
+"""A fully parallel Dimension that would benefit from tiling (or "blocking")."""
+
+WRAPPABLE = Property('wrappable')
+"""
+A modulo-N Dimension (i.e., cycling over i, i+1, i+2, ..., i+N-1) that could
+safely be turned into a modulo-K Dimension, with K < N. For example:
+u[t+1, ...] = f(u[t, ...], u[t-1, ...]) --> u[t-1, ...] = f(u[t, ...], u[t-1, ...]).
```

Comment:
that's `t-1` on both side of `=` in the second part.

Reply:
yes, that's the point. A very special case in which you could get away with two timeslots instead of three (here it's t-1 and t, could have been t+1 and t)

Reply:
If it's inside `f` it's note the case as it implies it's not just a overwrite but need to know `t-1` everywhere.